eva_submission/submission_qc_checks.py

```diff
@@ -0,0 +1,216 @@
+
+    @retry(tries=3, delay=2, backoff=1.5, jitter=(1, 3))
+    def get_result_from_webservice(self, url):
+        return requests.get(url)
+
+    def check_if_study_present_in_response(self, res, key, project_accession):
+        if any(res) and 'response' in res and len(res['response']) > 0:
+            for response in res['response']:
+                if response['numTotalResults'] >= 1:
+                    for result in response['result']:
+                        if result[key] == project_accession:
+                            return True
+        return False
+
+    def get_species_name(self, taxonomy, assembly, profile, private_config_xml_file):
+        with get_metadata_connection_handle(profile, private_config_xml_file) as pg_conn:
+            query = f"""select concat(t.taxonomy_code, '_',a.assembly_code) from evapro.taxonomy t 
+                        join evapro.assembly a on a.taxonomy_id = t.taxonomy_id 
+                        where t.taxonomy_id = {taxonomy} and assembly_accession='{assembly}'"""
+            return get_all_results_for_query(pg_conn, query)[0][0]
+
+    def get_browsable_files_for_study(self, profile, private_config_xml_file, project_accession):
+        with get_metadata_connection_handle(profile, private_config_xml_file) as pg_conn:
+            query = f"select filename from evapro.browsable_file where project_accession='{project_accession}'"
+            return [filename for filename, in get_all_results_for_query(pg_conn, query)]
+
+    def check_all_browsable_files_are_available_in_ftp(self, taxonomy, project_accession, vcf_files):
+        logger.info(f'Browsable files in db for study {project_accession}: {vcf_files}')
+
+        try:
+            files_in_ftp = self.get_files_from_ftp(project_accession)
+        except Exception as e:
+            logger.error(f'Error fetching files from ftp for study {project_accession}. Error {e}')
+            return
+
+        if not files_in_ftp:
+            logger.error(f'No files found in ftp for study {project_accession}')
+            return
+
+        logger.info(f'Files in ftp for study {project_accession}: {files_in_ftp}')
+
+        for file in vcf_files:
+            if file not in files_in_ftp:
+                logger.error(f'{file} not found in ftp')
+            if f'{file}.tbi' not in files_in_ftp:
+                logger.error(f'{file}.tbi not found in ftp')
+            if f'{file}.csi' not in files_in_ftp:
+                logger.error(f'{file}.csi not found in ftp')
+
+            # accessioned files will not be present for human taxonomy
+            if taxonomy != 9606:
+                accessioned_file = file.replace('.vcf.gz', '.accessioned.vcf.gz')
+                if accessioned_file not in files_in_ftp:
+                    logger.error(f'{accessioned_file} not found in ftp')
+                if f'{accessioned_file}.tbi' not in files_in_ftp:
+                    logger.error(f'{accessioned_file}.tbi not found in ftp')
```

Comment:
I don't think we generate tbi indices anymore (for both accessioned and submitted files).

Reply:
removed checks for **.tbi** files